### PR TITLE
Create util and add jump host

### PR DESCRIPTION
**Pull request by tybruno**
Added the jump_host context manager and StatfulAuthHandler. Also added documentation. Feel free to brutally give criticism. I will add more to the StatfulAuthHandler later to enable duo 2FA options.

### CODE REVIEW

File: paramiko_jump/util.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    Attributes
+    ----------
+    _iterator: Iterator
+        Iterator to iterate through all the objects in an iteratable object
+    """
+
+    _iterator = None
+
+    def __init__(self, *items: Sequence):
```
Comment (andrewschenck) [inline paramiko_jump/util.py:46]: Don't annotate items as type Sequence -- if anything, it should be type Any. The reason for this is that an annotation of a positional argument is an annotation of what is being passed in, not of the positional type itself (this isn't an annotation for items, but rather what items contains).

File: paramiko_jump/util.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        self._iterator = iter(items)
+
+    def __call__(self, *args: Sequence, **kwargs: Sequence):
```
Comment (andrewschenck) [inline paramiko_jump/util.py:49]: Same here on the annotations. I recommend dropping annotations on args/kwargs. They're idiomatic Python as-is without annotations.

**Comment by tybruno** [comment]
Made the changes you suggested

File: paramiko_jump/util.py (hunk 1 above)
Comment (tybruno) [reply paramiko_jump/util.py:44]: aaahhh ok that makes sense! thank you!

**Pull request merged**